src/interfaces/siteTree.ts

```diff
@@ -28,6 +32,9 @@ export const SiteTreeReadOnly : SiteTreeConstructor = function(menus) {
     const notCustomItemsByUrl : { [fullUrl : string]: MenuEntryByUrl } = {};
     const customItemsByUrl : { [fullUrl : string]: MenuEntryByUrl } = {};
     const levelZeroByUrl : { [urlSiteWithoutHomePage : string]: MenuEntryByUrl } = {};
+    const menusEntriesByObjectType: {[objectType: string] : { urlInstanceRestUrl: string, entries: MenuEntry }[]} = {}
+
+    info(`START ANALYSE`, { method: 'SiteTreeReadOnly' });
```

Comment:
(For later) Please account this in a Prometheus-friendly way i.e. increment some counter or histogram with the amount of time you spent preparing the data structures.